### DISCUSSION TITLE
need help: There are some issues when segmenting videos

### DISCUSSION THREAD

**Discussion by pengyouak** [Breakthrough/PySceneDetect#453] [Troubleshooting]
How to filter out short clips when segmenting videos?
How to re crop the beginning or end of each segment?

**Comment by Breakthrough** [comment]
> How to filter out short clips when segmenting videos?

You can do this pretty easily with a list comprehension:

```python
min_scene_len = 0.5
scenes = [(start, end) for (start, end) in scenes if (end - start) >= min_scene_len]

> How to re crop the beginning or end of each segment?

Could you clarify what you mean by this?

**Reply by pengyouak** [reply]
![image](https://github.com/user-attachments/assets/a8cceec6-0406-4cef-b802-d2cf25a2bb7d)

my english is not good , so i drawn an image to explain what did i mean.

**Reply by Breakthrough** [reply]
You would need to write the Python code to iterate over the scenes and trim the timecodes.  You can use the +/- operators to add floats representing seconds to the scenes.

For example to extend each scene by 0.5s:

    scenes = [(start, end + 0.5) for (start, end) in scenes if (end - start) >= min_scene_len]

**Reply by pengyouak** [reply]
非常感谢，我重新调整一下我的代码。
